lib/server/swarm.js

```diff
@@ -98,15 +108,18 @@ Swarm.prototype._onAnnounceUpdate = function (params, peer) {
     this.complete += 1
     this.incomplete -= 1
     peer.complete = true
+    var id = params.type === 'ws' ? params.peer_id : params.addr
+    this.peers.set(id, peer)
   }
 }
 
 Swarm.prototype._getPeers = function (numwant, ownPeerId, isWebRTC) {
   var peers = []
-  var ite = randomIterate(Object.keys(this.peers))
+  var ite = randomIterate(Object.keys(this.peers.cache))
   var peerId
   while ((peerId = ite()) && peers.length < numwant) {
-    var peer = this.peers[peerId]
+    // Don't mark the peer as most recently used on announce
+    var peer = this.peers.peek(peerId)
```

Comment:
There is a problem here, on one hand, if I use `LRU.get`, the peer will be marked as used and the `max` condition can evict a more recent peer.
On the other hand, if I use `LRU.peek`, we never access old peers and the `maxAge` eviction will not happen.

I think I will clean old peers manually as they are picked in announce...